pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -143,7 +143,19 @@ func (c *Controller) Services() ([]*model.Service, error) {
 					sp.ClusterVIPs = make(map[string]string)
 				}
 				sp.ClusterVIPs[r.ClusterID] = s.Address
-				smap[s.Hostname] = sp
```

Comment:
Doesn’t this break the existing multi cluster stuff @sdake?

Reply:
This `smap[s.Hostname] = sp` was redundant if you examine the current logic